certification project

Kraskodasha opened the pull request:
Написан скрипт для базового уровня

Kraskodasha committed:
certification project

Написан скрипт для базового уровня

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,230 @@
+  const card = document.createElement('div');
+  card.classList.add('card');
+  card.textContent = text;
+  card.dataset.type = type; // Добавляем тип карточки (foreign или translation)
```
Даша,  а где ты добавляешь dataset.value и dataset.pair, которые дальше используешь ? строки 191-192

 const wordObj = words.find(w => w.foreign === text || w.translation === text);
  if (type === 'foreign') {
    card.dataset.value = wordObj.foreign;
    card.dataset.pair = wordObj.translation;
  } else {
    card.dataset.value = wordObj.translation;
    card.dataset.pair = wordObj.foreign;
  }

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,230 @@
+      }, 1000);
+    }
+  }
+}
```
```suggestion
}
function handleCardClick(card) {

  if (card.classList.contains('fade-out')) return
  selectedCards.push(card);
 

  if (selectedCards.length === 1) {
    card.classList.add('correct');
    return;
  }

  if (selectedCards.length === 0) {
    card.classList.add('correct');
  }

  if (selectedCards.length === 2) {
    const [firstCard, secondCard] = selectedCards;

    if (firstCard.dataset.pair === secondCard.dataset.value &&
      secondCard.dataset.pair === firstCard.dataset.value) {
      // Карточки совпали
      firstCard.classList.add('correct');
      secondCard.classList.add('correct');
      selectedCards = [];
      matchedPairs++;

      setTimeout(() => {
        firstCard.classList.add('fade-out');
        secondCard.classList.add('fade-out');
      }, 500);

      // Проверяем завершение игры
      if (matchedPairs === words.length) {
        setTimeout(() => {
          alert('Игра успешно завершена!');
        }, 1000);
      }
    } else {
      // Если пара не совпала
      secondCard.classList.add('wrong');
      setTimeout(() => {
        firstCard.classList.remove( 'correct');
        secondCard.classList.remove( 'wrong');
        selectedCards = [];
      }, 1000);
    }
  }
}
```

Kraskodasha committed:
final project

Доработана функция клика по карточкам в тестовом режиме. Удалены ненужные классы. Добавлен обработчик события при клике на кнопку "Тестирование". Исправлены недочету по всему коду, в соответствии с установленными критериями.
